client/src/components/donut.tsx

```diff
@@ -1,22 +1,36 @@
 import React from 'react';
 import Base from './base';
 
-export default class Donut extends Base {
+type Props = {
+    percent: number;
+    percent2: number;
+}
+
+type State = {
+    currentPercent: number,
+    currentPercent2: number,
+};
+export default class Donut extends Base<Props, State> {
     state = {
         currentPercent: 0,
         currentPercent2: 0,
     };
 
-    constructor(props) {
+    private unmounted: boolean = false;
+
+    private animationQueued: boolean = false;
+
+    constructor(props: Props) {
         super(props);
         requestAnimationFrame(() => this.animate());
     }
 
-    componentWillUnmount() {
+    async componentWillUnmount() {
+        await this.clearDisposers();
```

Comment:
Great catch! But, it might be safer to call `super()` here instead incase we ever add more logic to the base.

Reply:
Good idea.  I will get that done real quick.

Reply:
Updated